.azure-pipelines-steps.yml

```diff
@@ -0,0 +1,27 @@
+steps:
+- script: npm i -g npm@$(npm_version)
+  displayName: Use legacy npm version $(npm_version)
```

Comment:
The node versions and the npm versions are aligned?  In travis and appveyor, when we "checkout" an node version, we get the distributed npm version

Reply:
We don't do that by default yet, I think I agree that we should have this as an option going forward though. If we want to make sure we have exact alignment we can run this step for every job for now.

I'm going to be working on some of our node tooling over the next couple of weeks, I'll try to make this better.

Reply:
Actually, did a little more digging on this. It turns out in the general case, we behave like we should (installing the appropriate version of npm). The only exception to this is on older versions of windows. This is because node doesn't automatically package them in their distribution in the same way. I still think it should be improved, but it only affects node<5 on Windows I think.

In the meantime, the current legacy install step should be sufficient to make sure the packages match the distribution so we don't need to make any immediate changes here.

Reply:
@damccorm thanks for looking into that!  I actually remember the "bad old days" when npm didn't ship with the Windows distribution.  Once we start dropping old versions (a ways off), we'll be able to reduce these phases.